components/server/src/jobs/workspace-gc.ts

```diff
@@ -80,16 +90,21 @@ export class WorkspaceGarbageCollector implements Job {
                     this.config.workspaceGarbageCollection.chunkLimit,
                 );
             const afterSelect = new Date();
-            const deletes = await Promise.all(
-                workspaces.map((ws) => this.workspaceService.deleteWorkspace(SYSTEM_USER, ws.id, "gc")),
-            );
+            log.info(`workspace-gc: about to soft-delete ${workspaces.length} workspaces`);
+            for (const ws of workspaces) {
+                try {
+                    await this.workspaceService.deleteWorkspace(SYSTEM_USER, ws.id, "gc");
+                } catch (err) {
+                    log.error({ workspaceId: ws.id }, "workspace-gc: error during workspace soft-deletion", err);
```

Comment:
Does it make sense to count failures and subtract from total?

Reply:
no, I don't think so. we will see the error logs in that case